src/testRunner/unittests/helpers/virtualFileSystemWithWatch.ts

```diff
@@ -1158,7 +1158,12 @@ export class TestServerHost implements server.ServerHost, FormatDiagnosticsHost,
 
         // base folder has to be present
         const base = getDirectoryPath(file.path);
-        const folder = Debug.checkDefined(this.getRealFolder(base));
+        const folder = this.getRealFolder(base);
+        // this line can throw on purpose, don't use Debug.assert to avoid redundant debugger hits
+        // eslint-disable-next-line no-restricted-syntax
+        if (folder === null || folder === undefined) {
+            throw new Error(`Directory not found: ${base}`);
+        }

```

Comment:
Do you mind sending this as a separate change; I've been wanting to do this for a while so would be nice to get in regardless of the status of this PR.

No need to leave the comment, and I suspect that a plain truthiness check is fine too.

Reply:
https://github.com/microsoft/TypeScript/pull/60009